### PR TITLE
feat: Notecard-only power outage monitoring

### PR DESCRIPTION
**Pull request by m-mcgowan**
A simple guide on using a Notecard to monitor mains power outages.

### CODE REVIEW

**Comment by m-mcgowan** [comment]
[NF16 - Power Outage Monitor](https://trello.com/c/CRulN0PZ/145-nf16-power-outage-monitor)

File: power-outage-monitor/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+## System Test
+
+In this section, you will power the Notecard via the power brick and simulate a power failure.
+
+### Setup
+
+1. Insert the power brick into a suitable electrical outlet and provide power to the Notecarrier via the micro USB cable.
+
+1. Wait for the Notecard to connect to the cloud.  When connected, there will be a new session event in Notehub. (look for recent `_session.qo` events.)
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:83]: This should be 2 and below should be 3.

**Review by haydenroche5 — CHANGES_REQUESTED** [review]
Did a quick read through of this, as I'll need to do the same thing in my refrigerator monitor app. Great documentation.

File: power-outage-monitor/README.md (hunk 1 above)
Comment (m-mcgowan) [reply power-outage-monitor/README.md:83]: Markdown renderers will number the lists correctly. https://github.com/blues/app-accelerators/tree/mat-power-outage-monitor/power-outage-monitor. 

Of course, if devs are reading markdown from the source it's a little more confusing, but really, do people do that? ;-)

Here's another example. https://raw.githubusercontent.com/blues/app-accelerators/main/low-power-digital-signage/README.md.  The reason for doing this is to make later revisions easier (e.g. inserting a new item in the middle of the list without having to renumber everything.)

Comment (haydenroche5) [reply power-outage-monitor/README.md:83]: TIL!

File: power-outage-monitor/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+## Hardware Setup
+
+1. Assemble Notecard, Notecarrier and LiPo battery as described in our [tutorial](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-b/).
+
+2. Keep the Notecard connected to the computer with the in-browser terminal active until you have completed [Notecard Setup](#sotecard-setup).
+
+## Notecard Firmware
+
+Your notecard should be running [firmware version 3.5.1](https://dev.blues.io/notecard/notecard-firmware-updates/#v3-5-1-october-7th-2022) or higher version. The simplest way to update firmware is to do an [over-the-air update](https://dev.blues.io/notecard/notecard-firmware-updates/#ota-dfu-with-notehub)
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:35]: "or higher version" -> "or higher"

File: power-outage-monitor/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+## Cloud Setup
+
+- Sign up for a free account on [Notehub.io](https://notehub.io), and create a new project.
+  `{"req":"hub.set", "product":"com.your-company.your-name:your_product", "mode":"continuous"}`
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:40]: Should this be here? We later ask the user to set it to periodic mode.

File: power-outage-monitor/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+## Notecard Setup
+
+In these steps, you will use the in-browser terminal to configure your notecard to report power outage events to your Notehub project.
+
+### Configure Notehub Project and Connection Mode
+
+First, we will set the serial number (name) of your notecard and the ProductUID. The ProductUID is the one corresponding to the project you careted in [Cloud Setup](#cloud-setup). Replace the `sn` and `product` values with your own and paste the resulting command into the in-browser terminal.
+
+```json
+{
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:52]: Just curious, are you able to paste multi-line requests in the in-browser terminal? I don't think it works for me.

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: power-outage-monitor/README.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
+# Power Outage Detector
+
+Use Notecard with a mains-powered USB power brick and LiPo battery to build a globally availalbe power outage detector.
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:3]: availalbe -> available

File: power-outage-monitor/README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+* USB A to Micro USB cable
+* [Notecarrier-A](https://shop.blues.io/collections/notecarrier/products/carr-al) or [Notecarrier-B](https://shop.blues.io/collections/notecarrier/products/carr-b)
+* Cellular [Notecard](https://blues.io/products/notecard/)
+
+
+## Hardware Setup
+
+1. Assemble Notecard, Notecarrier and LiPo battery as described in our [tutorial](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-b/).
+
+2. Keep the Notecard connected to the computer with the in-browser terminal active until you have completed [Notecard Setup](#sotecard-setup).
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:31]: sotecard -> notecard

File: power-outage-monitor/README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  `{"req":"hub.set", "product":"com.your-company.your-name:your_product", "mode":"continuous"}`
+- Twillio route: TBD
+
+## Notecard Setup
+
+In these steps, you will use the in-browser terminal to configure your notecard to report power outage events to your Notehub project.
+
+### Configure Notehub Project and Connection Mode
+
+First, we will set the serial number (name) of your notecard and the ProductUID. The ProductUID is the one corresponding to the project you careted in [Cloud Setup](#cloud-setup). Replace the `sn` and `product` values with your own and paste the resulting command into the in-browser terminal.
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:49]: careted -> created

File: power-outage-monitor/README.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+### Configure Power Outage Alerts
+
+Paste this command into the in-browser terminal
+
+```json
+{ "req": "card.voltage", "mode": "lipo", "usb": true, "alert": true, "sync": true }
+```
+
+> This instructs the Notecard to assume LiPo battery characterisitcs, monitor USB power and send an alert immediately when a power outage occurs and when power is restored. For more details, see the [Notecard API reference](https://dev.blues.io/reference/notecard-api/card-requests/#card-voltage) and the `Enable USB Power Alerting` example.
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:72]: characterisitcs -> characteristics

File: power-outage-monitor/README.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+1. Wait for the Notecard to connect to the cloud.  When connected, there will be a new session event in Notehub. (look for recent `_session.qo` events.)
+
+1. Wait for the battery to fully charge.
+
+### Simulating a Power Outage
+
+1. Remove the power brick from the electrical outlet to simulate a power outage. The Notecard is now being powered by the battery.
+
+2. After short time, you will see a  `_health.qo` event in Notehub with the body
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:91]: After short time -> After a short time

File: power-outage-monitor/README.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+```json
+{"text":"USB power OFF {usb-disabled}"}
+```
+
+This event indicates that the Notecard is no longer powered from the power brick.
+
+3. Reinsert the power brick into a suitable electrical outlet to simulate power being restored.
+
+4. After short time, you will see a  `_health.qo` event in Notehub with the body
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:101]: After short time -> After a short time

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: power-outage-monitor/README.md (hunk 4 above)
Comment (m-mcgowan) [reply power-outage-monitor/README.md:52]: yes it does work - but you can only paste on request at a time. The terminal removes the line breaks and sends the pasted text as a single request. I like your approach with the notecard CLI.

File: power-outage-monitor/README.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+```json
+{"text":"USB power ON {usb-enabled}"}
+```
+
+This event indicates that the Notecard is again powered by the power brick.
+
+
+### Blues Wireless Community
+
+We’d love to hear about you and your project on the [Blues Wireless Community Forum].
```
Comment (haydenroche5) [inline power-outage-monitor/README.md:120]: Should we move the link below here? i.e.:

```
[Blues Wireless Community Forum](https://discuss.blues.io/)
```

Comment (m-mcgowan) [reply power-outage-monitor/README.md:120]: I don't mind either way. I copied it from some existing docs in the repo.

Comment (haydenroche5) [reply power-outage-monitor/README.md:120]: I'll leave it up to you. I'll go ahead and give my seal of approval on the overall PR.

**Review by haydenroche5 — APPROVED** [review]

**Pull request merged**